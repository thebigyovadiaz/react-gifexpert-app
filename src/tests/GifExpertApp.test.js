import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas Unitarias de <GifExpertApp />', () => {
  test('Debe generarse el snapshot del componente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar una lista de categorias', () => {
    const defaultCategories = ["One Punch", "Dragon Ball Z"];
    const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories} />)
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
  });
});