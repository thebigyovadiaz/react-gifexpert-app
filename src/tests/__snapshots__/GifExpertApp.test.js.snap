// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Unitarias de <GifExpertApp /> Debe generarse el snapshot del componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas Unitarias de <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;
